File: juzgo/juzgo/custom/js/employee.js
frappe.ui.form.on("Employee",{
    create_user:function(frm){
        var d = new frappe.ui.Dialog({
            title: 'Profile Details',
            size: 'small',
            minimizable: true,
            static: true,
            fields: [
                {
                    label:"Role Profile",
                    fieldname: 'role_profile_name',
                    fieldtype: 'Link',
                    options: "Role Profile",
                },
                {
                    label:"Module Profile",
                    fieldname: 'module_profile',
                    fieldtype: 'Link',
                    options: "Module Profile",
                },
                {
                    label:"Password",
                    fieldname: 'new_password',
                    fieldtype: 'Password',
                }
            ],
            primary_action_label: 'Update User',
            secondary_action_label: 'Close',
            async primary_action(data) {
                if (!frm.doc.prefered_email) {
                    frappe.throw(__("Please enter Preferred Contact Email"));
                }
                frappe.call({
                    method: "juzgo.juzgo.custom.py.employee.create_users",
                    args: {
                        employee: frm.doc.name,
                        email: frm.doc.prefered_email,
                        data:data
                    },
                    callback: function (r) {
                        frm.set_value("user_id", r.message);
                        frm.save() //corecodewhile creating a user employee can save the auto
                    }
                });
                d.hide()
            },
            secondary_action() {
                d.hide()
            }
        })
        d.$wrapper.find('.btn-link').css('width', '100px');
        d.show()
    }
})

File: juzgo/juzgo/custom/py/employee.py
import json
import frappe

@frappe.whitelist()
def create_users(employee, user=None, email=None, data=None):
	emp = frappe.get_doc("Employee", employee)
	data = json.loads(data)
	employee_name = emp.employee_name.split(" ")
	middle_name = last_name = ""

	if len(employee_name) >= 3:
		last_name = " ".join(employee_name[2:])
		middle_name = employee_name[1]
	elif len(employee_name) == 2:
		last_name = employee_name[1]

	first_name = employee_name[0]

	if email:
		emp.prefered_email = email

	user = frappe.new_doc("User")
	user.update(
		{
			"name": emp.employee_name,
			"email": emp.prefered_email,
			"enabled": 1,
			"first_name": first_name,
			"middle_name": middle_name,
			"last_name": last_name,
			"gender": emp.gender,
			"birth_date": emp.date_of_birth,
			"phone": emp.cell_number,
			"bio": emp.bio,
            "new_password":data.get("new_password"),
            "module_profile":data.get("module_profile"),
            "role_profile_name":data.get("role_profile_name")
		}
	)
	user.insert()
	return user.name